costumes.js

```diff
@@ -0,0 +1,110 @@
+const createCostume = async (
+    pool,
+    {
+        name, 
+        category, 
+        gender, 
+        size, 
+        type, 
+        stockCount, 
+        price
+    }) => {
+    const {rows:[costume]} = await pool.query(`
+        INSERT INTO costumes(
+            name,
+            category,
+            gender,
+            size,
+            type,
+            stock_count,
+            price
+        ) VALUES ($1, $2, $3, $4, $5, $6, $7)
+        RETURNING *;   
+    `, [name, category, gender, size, type, stockCount, price]);
+    return costume;
+}
+
+const getAllCostumes = async (pool) => {
+    const {rows: costumes} = await pool.query(`
+        SELECT * FROM costumes;
+    `)
+    return costumes;
+}
+
+const getCostumeById = async (pool, id) => {
+    const {rows:[costume]} = await pool.query(`
+        SELECT *
+        FROM costumes
+        WHERE id = $1;
+    `, [id])
+    if (costume === undefined) {
+        throw new Error(`Could not retrieve data because id provided (${id}) does not exist in table.`)
+    } 
+    return costume;
+}
+
+const updateCostume = async (
+    pool,
+    id,
+    {
+        name, 
+        category,
+        gender,
+        size,
+        type,
+        stockCount,
+        price
+    }) => {
+    const {rows: [costume]} = await pool.query(`
+        UPDATE costumes
+        SET 
+            name = $1,
+            category = $2,
+            gender = $3,
+            size = $4,
+            type = $5,
+            stock_count = $6,
+            price = $7
+        WHERE id = $8
+        RETURNING *;
+    `, [
+        name, 
+        category, 
+        gender, 
+        size, 
+        type, 
+        stockCount, 
+        price, 
+        id
+    ])
+    if (costume === undefined) {
+        throw new Error(`Could not update row because id provided (${id}) does not exist in table.`)
+    } 
+    return costume;
+}
+
+const deleteCostumeById = async (pool, id) => {
+    const {rows: costumes} = await pool.query(`
+        SELECT * FROM costumes 
+        WHERE id = $1;
+    `, [id])
+    
+    if (costumes.length === 0) {
+        throw new Error(`Could not delete row because id provided (${id}) does not exist in table.`)
+    } 
```

Comment:
Is this necessary? What will happen if you try to delete a costume that doesn't exist?

Reply:
Right you are! This error is thrown on its own: `Error: Could not delete row because id provided (3) does not exist in table.`

Reply:
It does nothing and no error is thrown.

Reply:
https://www.postgresql.org/docs/current/sql-delete.html

> The optional RETURNING clause causes DELETE to compute and return value(s) based on each row actually deleted. 

I think this is the best course of action. Do RETURNING at the end, and then just check if the returned costume was null

By the way, since it requires a RETURNING to return something, I'm not sure how this error was working and not breaking tests? Wouldn't it always have length 0 if there's no RETURNING clause? Maybe it returns the object by default?

Reply:
Without a RETURNING clause, it was returning an array of objects representing the rows deleted. If nothing was deleted, it returned an empty array.

Reply:
Just checked and it does the same thing with a RETURNING clause on DELETE. I changed my code to use destructuring assignment for the element that is in the array returned by the DELETE query, that way I can do a truthy check instead of checking the length property of the array. With that setup, it returns `undefined` if nothing was deleted.